### PR TITLE
coreboot version bump from 4.22.01 to 24.02.01 (Q35/ivy/sandy/haswell/librems) 

### PR DESCRIPTION
**Pull request by tlaurion**
Coreboot version bump to 24.02.01 for all platforms that depended on coreboot 4.20.01. 

**Testable roms available under green mark, select board, then go under artifacts and select zip file to internally upgrade. Do backup first from recovery shell and be ready to externally flash backup if bricked.** 

EDIT: [8d3c5cf](https://github.com/linuxboot/heads/pull/1715/commits/8d3c5cf355d8d2777dd4ab10c7cf95f2d1f9c7f1) inclues @JonathonHall-Purism patches (that can be seen under `patches/coreboot-24.02.01` of this PR):

Tested:

- [ ] qemu tpm1/tpm2: ok!
- [ ] x230

-----
Old (prior of [8d3c5cf](https://github.com/linuxboot/heads/pull/1715/commits/8d3c5cf355d8d2777dd4ab10c7cf95f2d1f9c7f1))

Tested
- [ ] t430 : thanks @aluciani https://github.com/linuxboot/heads/pull/1715#issuecomment-2222381749 **NO BOOTSPLASH** 
- [ ] x230 (@tlaurion) : https://output.circle-artifacts.com/output/job/a3a29799-197a-43a6-ad76-ec3b0e1b238d/artifacts/0/build/x86/x230-maximized/heads-x230-maximized-v0.2.0-2215-g21ab781.zip **BOOTSLASH IS GONE**

----
All boards untested will be prepended UNTESTED, then all boards UNTESTED will go UNMAINTAINED.

### CODE REVIEW

**Comment by tlaurion** [comment]
@aluciani roms have built successfully. T430 can be tested and results posted here.

**Comment by aluciani** [comment]
> * https://output.circle-artifacts.com/output/job/fbb7519a-41cd-4a60-a714-d18604b40c36/artifacts/0/build/x86/t430-maximized/heads-t430-maximized-v0.2.0-2215-g21ab781.zip

working on my t430

**Comment by aluciani** [comment]
> Tested https://output.circle-artifacts.com/output/job/a3a29799-197a-43a6-ad76-ec3b0e1b238d/artifacts/0/build/x86/x230-maximized/heads-x230-maximized-v0.2.0-2215-g21ab781.zip working (not causing brick).
> 
> **BOOTSPLASH IS GONE**

I was going to report 
The t430 also miss the bootsplash, but booting to debian 12 is working, didn t really did extensive test ...

**Comment by tlaurion** [comment]
Putting back as draft: cbmem -c | grep -ie error -ie warning shows errors with memalign an dout of memory when trying to map bootsplash.

Also shows PNP: missing read resources errors.

**Comment by tlaurion** [comment]
Testing local builds pointing to v0.2.0 docker image which contains libnss3-dev new coreboot requirement from built docker image from https://github.com/linuxboot/heads/pull/1715/commits/21ab781e901d619f1a15f543fb6d1477fb5f458b which was https://github.com/linuxboot/heads/pull/1713

**Comment by tlaurion** [comment]
`docker run -e DISPLAY=$DISPLAY --network host --rm -ti -v $(pwd):$(pwd) -w $(pwd) tlaurion/heads-dev-env:v0.2.0 -- make BOARD=qemu-coreboot-whiptail-tpm1`

Replicates the no bootsplash issue. So can be tested with qemu, which will be more easy to report to coreboot and involve coreboot to fix this without real hardware.

**Comment by tlaurion** [comment]
> `docker run -e DISPLAY=$DISPLAY --network host --rm -ti -v $(pwd):$(pwd) -w $(pwd) tlaurion/heads-dev-env:v0.2.0 -- make BOARD=qemu-coreboot-whiptail-tpm1`
> 
> Replicates the no bootsplash issue. So can be tested with qemu, which will be more easy to report to coreboot and involve coreboot to fix this without real hardware.
```
~ # cbmem -c | grep -ie error -ie warning
[ERROR]  memalign(boundary=8, size=1179648): failed: Tried to round up free_mem_ptr 0x7fece6c8 to 0x7ffee6c8
[ERROR]  but free_mem_end_ptr is 0x7ffcc7d0
[ERROR]  Error! memalign: Out of memory (free_mem_ptr >= free_mem_end_ptr)Bootsplash could not be decoded. jpeg_decode returned 1.
```

@JonathonHall-Purism I guess you get the same on librems?

**Comment by tlaurion** [comment]
Attempting to build for coreboot 2405 fils with coreboot buildstack dependencies pointing to unavailable tarballs, as can be seen with this CircleCI build attempt at https://app.circleci.com/pipelines/github/tlaurion/heads/2642/workflows/92c0c23d-340a-4bd5-8cd7-9b5b479b00bf for testing branch https://github.com/tlaurion/heads/tree/xxx0_coreboot2405 testing commit https://github.com/tlaurion/heads/commit/481c17326e359d3e9d191a715b276ed8e28d0828 with commit log stating error that will show in build logs there....


coreboot regressions all around it seems.

**Comment by tlaurion** [comment]
> Attempting to build for coreboot 2405 fils with coreboot buildstack dependencies pointing to unavailable tarballs, as can be seen with this CircleCI build attempt at https://app.circleci.com/pipelines/github/tlaurion/heads/2642/workflows/92c0c23d-340a-4bd5-8cd7-9b5b479b00bf for testing branch https://github.com/tlaurion/heads/tree/xxx0_coreboot2405 testing commit [tlaurion@481c173](https://github.com/tlaurion/heads/commit/481c17326e359d3e9d191a715b276ed8e28d0828) with commit log stating error that will show in build logs there....
> 
> coreboot regressions all around it seems.
> 
> TLDR: coreboot 2405 depends on https://distfiles.macports.org/acpica/acpica-unix-20230628.tar.gz which doesn't exist. See: https://app.circleci.com/pipelines/github/tlaurion/heads/2642/workflows/92c0c23d-340a-4bd5-8cd7-9b5b479b00bf/jobs/48310/parallel-runs/0/steps/0-102

@JonathonHall-Purism can you cache https://downloadmirror.intel.com/783534/acpica-unix-20230628.tar.gz under purism mirror please?

**Comment by tlaurion** [comment]
> > Attempting to build for coreboot 2405 fils with coreboot buildstack dependencies pointing to unavailable tarballs, as can be seen with this CircleCI build attempt at https://app.circleci.com/pipelines/github/tlaurion/heads/2642/workflows/92c0c23d-340a-4bd5-8cd7-9b5b479b00bf for testing branch https://github.com/tlaurion/heads/tree/xxx0_coreboot2405 testing commit [tlaurion@481c173](https://github.com/tlaurion/heads/commit/481c17326e359d3e9d191a715b276ed8e28d0828) with commit log stating error that will show in build logs there....
> > coreboot regressions all around it seems.
> > TLDR: coreboot 2405 depends on https://distfiles.macports.org/acpica/acpica-unix-20230628.tar.gz which doesn't exist. See: https://app.circleci.com/pipelines/github/tlaurion/heads/2642/workflows/92c0c23d-340a-4bd5-8cd7-9b5b479b00bf/jobs/48310/parallel-runs/0/steps/0-102
> 
> @JonathonHall-Purism can you cache https://downloadmirror.intel.com/783534/acpica-unix-20230628.tar.gz under purism mirror please?

@JonathonHall-Purism I just came to realize that acpica-unix is not cached under packages/x86/...

**Comment by tlaurion** [comment]
> > > Attempting to build for coreboot 2405 fils with coreboot buildstack dependencies pointing to unavailable tarballs, as can be seen with this CircleCI build attempt at https://app.circleci.com/pipelines/github/tlaurion/heads/2642/workflows/92c0c23d-340a-4bd5-8cd7-9b5b479b00bf for testing branch https://github.com/tlaurion/heads/tree/xxx0_coreboot2405 testing commit [tlaurion@481c173](https://github.com/tlaurion/heads/commit/481c17326e359d3e9d191a715b276ed8e28d0828) with commit log stating error that will show in build logs there....
> > > coreboot regressions all around it seems.
> > > TLDR: coreboot 2405 depends on https://distfiles.macports.org/acpica/acpica-unix-20230628.tar.gz which doesn't exist. See: https://app.circleci.com/pipelines/github/tlaurion/heads/2642/workflows/92c0c23d-340a-4bd5-8cd7-9b5b479b00bf/jobs/48310/parallel-runs/0/steps/0-102
> > 
> > 
> > @JonathonHall-Purism can you cache https://downloadmirror.intel.com/783534/acpica-unix-20230628.tar.gz under purism mirror please?
> 
> @JonathonHall-Purism I just came to realize that acpica-unix is not cached under packages/x86/...

@JonathonHall-Purism that was our fault, we override the acpica-unix url defined under our mirroring using script which I hacked now as of https://github.com/tlaurion/heads/commit/71a60352e02feba35e15709381557d3bea15e6ae

**Comment by fhvyhjriur** [comment]
Would you like to send link to maximized T530, T420(dgpu - if you like to also test dgpu), and T440p(dgpu - if you like to also test dgpu) versions? I would take the risk on me to have to spend hours on each device externally reflashing it.

**Comment by fhvyhjriur** [comment]
I think i have find out the files for this PR. I expected same job-id in the link, but it seems like the job-id should differ from thinkpad model to other thinkpad model. When i am correct, this should be the image for T530: https://output.circle-artifacts.com/output/job/dc335a65-ce66-4455-ade6-298db6863f2a/artifacts/0/build/x86/UNMAINTAINED_t530-maximized/heads-UNMAINTAINED_t530-maximized-v0.2.0-2215-g21ab781.rom

Should i wait for the new release with the updated bootsplash jpg file and report if its working fine with the different screen resolutions? They differ from thinkpad to thinkpad and this would probably be a good test.

**Comment by tlaurion** [comment]
Asked coreboot devels at:
- matrix: https://matrix.to/#/!EhaGFZyYcbyhdSgStq:matrix.org/$5c2GtCoHQJg8NI_2bDeCJPQ_4OafdTRbOXwFRvn9e_s?via=matrix.org&via=sibnsk.net&via=nope.chat
- slack: https://osfw.slack.com/archives/CCWLQKEHG/p1721048604525559

TLDR of current issue under commit message at https://github.com/tlaurion/heads/commit/8a4ac331fc6c248038c986d767093068586744e9

**Comment by JonathonHall-Purism** [comment]
> @JonathonHall-Purism can you cache https://downloadmirror.intel.com/783534/acpica-unix-20230628.tar.gz under purism mirror please?

Mirror is updated with deps for 24.02.01 (this PR) and 24.05 (#1716).  Thanks for the reminder.

**Comment by tlaurion** [comment]
@JonathonHall-Purism I added your fixes, tested on qemu and works.

This invalidates #1720 which I will close. 
#1716 : not from now and will be based on this PR, so closing as well.

**Comment by tlaurion** [comment]
@JonathonHall-Purism posted RP to oreboot version bump to pr pointing to their coreboot fork with fixes included at https://github.com/linuxboot/heads/pull/1721

@JonathonHall-Purism : another PR will need to change CircleCI board cache dependencies: all board will be able to share 24.02.01 caches once both are merged.

**Comment by tlaurion** [comment]
> I think i have find out the files for this PR. I expected same job-id in the link, but it seems like the job-id should differ from thinkpad model to other thinkpad model. When i am correct, this should be the image for T530: https://output.circle-artifacts.com/output/job/dc335a65-ce66-4455-ade6-298db6863f2a/artifacts/0/build/x86/UNMAINTAINED_t530-maximized/heads-UNMAINTAINED_t530-maximized-v0.2.0-2215-g21ab781.rom
> 
> Should i wait for the new release with the updated bootsplash jpg file and report if its working fine with the different screen resolutions? They differ from thinkpad to thinkpad and this would probably be a good test.

@fhvyhjriur once [8d3c5cf](https://github.com/linuxboot/heads/pull/1715/commits/8d3c5cf355d8d2777dd4ab10c7cf95f2d1f9c7f1) CircleCI builds are successful, the roms produces for xx20/xx30/xx4x will need testing.

@JonathonHall-Purism Thanks for your precious contribution upstream!

**Comment by tlaurion** [comment]
@JonathonHall-Purism since PR title is also for Librems, I could merge your PR here so that another commit adds CircleCI boards dependencies to reduce time for ROM production if all boards sharing 24.02.01 could be built from same caches?

**Comment by tlaurion** [comment]
[7454d58](https://github.com/linuxboot/heads/pull/1715/commits/7454d587f8549cb96f66be5b7a6422230f7f150e) is merging changes from #1700 (merged to master)

**Comment by tlaurion** [comment]
@JonathonHall-Purism testing gains of shared coreboot 24.02.01 buildstack at https://github.com/tlaurion/heads/tree/test_240201_shared_coreboot_buildstack building at https://app.circleci.com/pipelines/github/tlaurion/heads/2667/workflows/5086fa75-30a6-44ac-aa72-69bf140de9c7

If works/useful: will add that commit here.
Testing on x230 and if successful, call for testing for coreboot version bump.

**Comment by fhvyhjriur** [comment]
> @fhvyhjriur once [8d3c5cf](https://github.com/linuxboot/heads/pull/1715/commits/8d3c5cf355d8d2777dd4ab10c7cf95f2d1f9c7f1) CircleCI builds are successful, the roms produces for xx20/xx30/xx4x will need testing.

Build failed.

**Comment by tlaurion** [comment]
> @JonathonHall-Purism testing gains of shared coreboot 24.02.01 buildstack at https://github.com/tlaurion/heads/tree/test_240201_shared_coreboot_buildstack building at https://app.circleci.com/pipelines/github/tlaurion/heads/2667/workflows/5086fa75-30a6-44ac-aa72-69bf140de9c7
> 
> If works/useful: will add that commit here. Testing on x230 and if successful, call for testing for coreboot version bump.

Fails since coreboot forks and coreboot releases tarballs cannot be mixed to cross use their buildstacks. This will be postponed.
Anyway: purism uses a newer kernel than xx20/xx30/xx4x/qemu, so community supported platforms (no funding, not dev time paid) needs to follow paid development fork to economize build time. In this case: purism uses a newer kerne then heads community supported boards, so gains of caching build artifacts would be lesser than ideal (cache needs to have prebuilt coreboot build stack and kernel artifacts to be fully efficient.

Will open issue and unrelated to this coreboot 24.02.01 being common. Will open issue to track and fix.

**Comment by tlaurion** [comment]
> > @fhvyhjriur once [8d3c5cf](https://github.com/linuxboot/heads/pull/1715/commits/8d3c5cf355d8d2777dd4ab10c7cf95f2d1f9c7f1) CircleCI builds are successful, the roms produces for xx20/xx30/xx4x will need testing.
> 
> Build failed for some boards. The build for the boards i have did not fail. For example this looks to be the build for T530: https://output.circle-artifacts.com/output/job/cf336210-0ac2-4458-82a0-a4bd8db904fb/artifacts/0/build/x86/UNMAINTAINED_t530-maximized/heads-UNMAINTAINED_t530-maximized-v0.2.0-2225-g9e9ab76.rom Should i test [8d3c5cf](https://github.com/linuxboot/heads/pull/1715/commits/8d3c5cf355d8d2777dd4ab10c7cf95f2d1f9c7f1) and ignore the failed build for some systems?

@aluciani there is no gain outside of sharing coreboot buildstack (crossgcc) and linux built artifacts to speed up CircleCI as said in previous comment, and as you witnessed, ti doesn't work out of the box today under CircleCI building 

> @JonathonHall-Purism testing gains of shared coreboot 24.02.01 buildstack at https://github.com/tlaurion/heads/tree/test_240201_shared_coreboot_buildstack building at https://app.circleci.com/pipelines/github/tlaurion/heads/2667/workflows/5086fa75-30a6-44ac-aa72-69bf140de9c7
> 
> If works/useful: will add that commit here. Testing on x230 and if successful, call for testing for coreboot version bump.


When it comes to testing PRs, when you scroll down, you will see all tests and links to CirlceCI to download roms.
Note again that for internal flashing, it is preferred to download zip files then rom files, since the zip file contains a sha256sum of the rom file and would prevent USB thumb drive flash corruption from causing a brick, so please use zip file for internal flashing instead of flashing rom files directly.

Note that this PR final commit (unless bugs are found under specific platforms) is [7454d58](https://github.com/linuxboot/heads/pull/1715/commits/7454d587f8549cb96f66be5b7a6422230f7f150e) which is atop of master, and completed producing rom/zipfiles artifacts for all boards currently defined under CircleCI under master.

**Comment by tlaurion** [comment]
I have put this PR as draft because i'm concerned in differences between defconfig changes between 4.22.01 and 24.02.01, and since doing a call for testing from all #692 was historically long, I would like this PR to contain proper coreboot changes to be tested by all community testers to fasten this merge as much as possible.

**Comment by tlaurion** [comment]
Something is corrupting heads repo from coreboot forks branch.

Traces of what I try to accomplish here.

Try to generate all coreboot defconfig to see deviation from coreboot specific version:
`grep -R CONFIG_COREBOOT_VERSION boards/ | awk -F "/" {'print $2'} | while read board; do if ! sudo make BOARD=$board coreboot.save_in_defconfig_format_in_place > /dev/null 2>&1; then echo $board failed;fi; done`

Then from failed boards, pick one board per shared coreboot version and build:
```
docker run -e DISPLAY=$DISPLAY --network host --rm -ti -v $(pwd):$(pwd) -w $(pwd) tlaurion/heads-dev-env:v0.2.0 -- make BOARD=nitropad-nv41
docker run -e DISPLAY=$DISPLAY --network host --rm -ti -v $(pwd):$(pwd) -w $(pwd) tlaurion/heads-dev-env:v0.2.0 -- make BOARD=nitropad-nv41
docker run -e DISPLAY=$DISPLAY --network host --rm -ti -v $(pwd):$(pwd) -w $(pwd) tlaurion/heads-dev-env:v0.2.0 -- make BOARD=qemu-coreboot-whiptail-tpm1
docker run -e DISPLAY=$DISPLAY --network host --rm -ti -v $(pwd):$(pwd) -w $(pwd) tlaurion/heads-dev-env:v0.2.0 -- make BOARD=librem_l1um
docker run -e DISPLAY=$DISPLAY --network host --rm -ti -v $(pwd):$(pwd) -w $(pwd) tlaurion/heads-dev-env:v0.2.0 -- make -d BOARD=talos-2

```

Then regenerate coreboot configs in defconfig:
grep -R CONFIG_COREBOOT_VERSION boards/ | awk -F "/" {'print $2'} | while read board; do if ! sudo make BOARD=$board coreboot.save_in_defconfig_format_in_place > /dev/null 2>&1; then echo $board failed;fi; done


--- 
Unfortunately, heads repo gets replaced by Dasharo when doing some randomly. Will revert to backup and snapshot qube in between to figure out what is happening with `make -d`, something seems wrong under modules/coreboot: picking wrong git root and replacing Heads instead of coreboot local dir which should be under `build/ARCH/coreboot-VER`

**Comment by tlaurion** [comment]
> Something is corrupting heads repo from coreboot forks branch.
> 
> Traces of what I try to accomplish here.
> 
> Try to generate all coreboot defconfig to see deviation from coreboot specific version:
> `grep -R CONFIG_COREBOOT_VERSION boards/ | awk -F "/" {'print $2'} | while read board; do if ! sudo make BOARD=$board coreboot.save_in_defconfig_format_in_place > /dev/null 2>&1; then echo $board failed;fi; done`
> 
> Then from failed boards, pick one board per shared coreboot version and build:
> ```
> docker run -e DISPLAY=$DISPLAY --network host --rm -ti -v $(pwd):$(pwd) -w $(pwd) tlaurion/heads-dev-env:v0.2.0 -- make BOARD=nitropad-nv41
> docker run -e DISPLAY=$DISPLAY --network host --rm -ti -v $(pwd):$(pwd) -w $(pwd) tlaurion/heads-dev-env:v0.2.0 -- make BOARD=nitropad-nv41
> docker run -e DISPLAY=$DISPLAY --network host --rm -ti -v $(pwd):$(pwd) -w $(pwd) tlaurion/heads-dev-env:v0.2.0 -- make BOARD=qemu-coreboot-whiptail-tpm1
> docker run -e DISPLAY=$DISPLAY --network host --rm -ti -v $(pwd):$(pwd) -w $(pwd) tlaurion/heads-dev-env:v0.2.0 -- make BOARD=librem_l1um
> docker run -e DISPLAY=$DISPLAY --network host --rm -ti -v $(pwd):$(pwd) -w $(pwd) tlaurion/heads-dev-env:v0.2.0 -- make -d BOARD=talos-2
> 
> ```
> 
> Then regenerate coreboot configs in defconfig:
> grep -R CONFIG_COREBOOT_VERSION boards/ | awk -F "/" {'print $2'} | while read board; do if ! sudo make BOARD=$board coreboot.save_in_defconfig_format_in_place > /dev/null 2>&1; then echo $board failed;fi; done
> 
> 
> --- 
> Unfortunately, heads repo gets replaced by Dasharo when doing some randomly. Will revert to backup and snapshot qube in between to figure out what is happening with `make -d`, something seems wrong under modules/coreboot: picking wrong git root and replacing Heads instead of coreboot local dir which should be under `build/ARCH/coreboot-VER`

This was me calling docker in parallel for multiple boards.

**Comment by tlaurion** [comment]
Deprecated in favor of https://github.com/linuxboot/heads/pull/1723
